src/components/pages/TeacherDashboard/TeacherViewSection/SectionStudentList/SectionStudentListContainer.tsx

```diff
@@ -18,8 +18,8 @@ const SectionStudentListContainer = ({
   });
 
   useEffect(() => {
-    getStudents(section.id);
-  }, [section]);
+    if (!studentList && !isLoading) getStudents(section.id);
```

Comment:
Can we check if this loads the correct students for the class? I think I was wrong we might need to revert this